src/yukon/backend/src/devserv/mock_server.py

```diff
@@ -0,0 +1,423 @@
+        return self._session_scheduler
+
+    @property
+    def event(self) -> ServerSentEvent:
+        return self._event
+
+    @event.setter
+    def event(self, event: ServerSentEvent) -> None:
+        self._event = event
+
+    def load_mock_system_description(self) -> None:
+        """
+        Loads mock system description from JSON file. The data is then split by
+        the diferent components and loaded once, with each response sent to its
+        specific path.
+        """
+        with open(os.path.join(dir_path, self.sys_descr + ".json")) as json_file:
+            data = json.load(json_file)
+
+            # Bus specific data. Gets rendered in the BusInfo component of the
+            # Home page
+            if 'bus' in data:
+                self.mock_response(os.path.join(
+                    self.api_prefix, 'bus'), 'bus_mock()', data['bus'])
+
+            # Node specific data. Gets rendered in the NodeList component of the
+            # Home page but also on each node specific detail page.
+            if 'nodes' in data:
+                if data['nodes']['detail']:
+                    # Rendered in the NodeList component
+                    self.mock_response(os.path.join(
+                        self.api_prefix, 'nodes'), 'nodes_info_mock()', data['nodes']['detail'])
+
+                    for nodes in data['nodes']['detail']:
+                        self.handle_nodes_detail(data, nodes)
+
+                # Rendered in the Plug&Play table in the Home page
+                # Note: currently hidden
+                if data['nodes']['plugandplay']:
+                    self.mock_response(os.path.join(
+                        self.api_prefix, 'nodes/plugandplay'), 'plugandplay_mock()', data['nodes']['plugandplay'])
+
+                # Rendered in the GlobalRegisterView page
+                # Note: needs further improvements
+                if data['nodes']['grv']:
+                    self.mock_response(os.path.join(
+                        self.api_prefix, 'nodes/grv'), 'grv_mock()', data['nodes']['grv'])
+
+            # Backend server health information
+            # Rendered in the ServerHealth component of the Home page
+            if 'health' in data:
+                self.mock_response(os.path.join(
+                    self.api_prefix, 'health'), 'health_mock()', data['health'])
+
+            # Data Type information
+            # Note: needs further improvement and assotiation to the PRDT
+            if 'types' in data:
+                self.mock_response(os.path.join(
+                    self.api_prefix, 'types'), 'types_mock()', data['types'])
+
+    def handle_nodes_detail(self, data: dict, nodes: dict) -> None:
+        """
+        Create mock response for node details.
+        """
+        # Node detailed info
+        # Rendered in each node page (identified by its ID)
+        self.mock_response(os.path.join(self.api_prefix, 'nodes/' + str(nodes['id'])),
```

Comment:
I'm not sure if this is a valid use of `os.path.join` because my understanding is that the path separator used here is independent of the platform-specific filesystem path separator; that is, running on Windows, you would still expect it to be `/` rather than `\`, is that not so?

Reply:
Pavel's right. You want to use some sort of url builder. I think the standard library has https://docs.python.org/3.8/library/urllib.parse.html?highlight=urljoin#urllib.parse.urljoin but the frameworks you are using might have specialized ones?

Reply:
Yeah no need to fancy things. Simple string concatenation is enough. Solved.